zipkin-storage/cassandra3/src/main/java/zipkin/storage/cassandra3/CassandraUtil.java

```diff
@@ -96,14 +97,17 @@ public static int durationIndexBucket(long ts_micro) {
       return Collections.emptyList();
     }
     checkArgument(request.serviceName != null, "serviceName needed with annotation query");
-    Set<String> annotationKeys = new LinkedHashSet<>();
```

Comment:
basically it is an edge case, where request.annotations has a key that's in request.binaryAnnotations.. perhaps we can just dedupe in the request object instead.. or not bother and switch to list with a risk of accidentally making the same query twice.. wdyt